src/Microsoft.VisualStudio.Editors/ResourceEditor/Resource.vb

```diff
@@ -1630,24 +1632,28 @@ Namespace Microsoft.VisualStudio.Editors.ResourceEditor
         ''' file are only used to produce the .designer.cs file, which is still built against
         ''' netstandard.dll.
         ''' </remarks>
-        Private Shared Function AdjustSystemStringAssemblyQualifiedName(AssemblyQualifiedName As String) As String
-            Static SystemStringAssemblyQualifiedName As String
-
-            ' If this type definitely isn't System.String then bail out before we allocate
-            If Not AssemblyQualifiedName.StartsWith("System.String") Then
-                Return AssemblyQualifiedName
-            End If
+        Private Shared Function AdjustAssemblyQualifiedName(AssemblyQualifiedName As String) As String
+            Dim CorrectedAssemblyQualifiedName As String = Nothing
 
             Dim indexOfFirstComma = AssemblyQualifiedName.IndexOf(",")
             If indexOfFirstComma <> -1 Then
                 Dim typeName = AssemblyQualifiedName.Substring(startIndex:=0, length:=indexOfFirstComma)
-                If typeName = "System.String" Then
-
-                    If SystemStringAssemblyQualifiedName Is Nothing Then
-                        SystemStringAssemblyQualifiedName = GetType(String).AssemblyQualifiedName
+                If s_correctedAssemblyQualifiedName.TryGetValue(typeName, CorrectedAssemblyQualifiedName) Then
+                    Return CorrectedAssemblyQualifiedName
+                Else
+                    If typeName = "System.String" Then
```

Comment:
To simplify this even further, just initialize the dictionary already populated with all the types and you can remove the if statements.